### DISCUSSION TITLE
Simplest way to deploy vaultwarden?

### DISCUSSION THREAD

**Discussion by NoniDOTio** [dani-garcia/vaultwarden#2435] [Q&A]
I am looking for an easy 1-click-solution to deploy the vaultwarden image from docker hub.

On my Plesk server, this was super simple using Plesks' Docker extension. Are there similar solutions for this? I tried Digitalocean Apps but it wouldn't let me deploy. Any suggestions on a host that would be suitable for this?

**Comment by RealOrangeOne** [comment]
Any managed docker deployment should work just fine. What didn't work about DO Apps?

There's very little in what you need which is vaultwarden specific. It's just an application with a database, that's all it needs.

The simplest way to run vaultwarden is to just use bitwarden.com.

**Reply by NoniDOTio** [reply]
Digitalocean deployment fails with this error message, which lead me to believe that these kinda simple setups don't work (they do not allow volume mapping which I used on Plesk)
```
[vaultwarden::util][WARN] Can't connect to database, retrying: DieselMig.
[CAUSE] QueryError(
     DatabaseError(
         __Unknown,
         "disk I/O error",
     ),
)
```
But I just tried on Azure and it seemed to work on there even without any volume mapping, so maybe it's just a DO-specific error?

**Reply by RealOrangeOne** [reply]
DO doesn't have persistent storage, no. So you'd need an external database
